tests/integration_tests/test_cli.py

```diff
@@ -286,6 +325,73 @@ def test_inspect( self ):
         cli.config = config
         cli.run()
 
+    def test_list( self ):
+        # Mock IO for wallet
+        with patch('bittensor.wallet.__new__', side_effect=[MagicMock(
+            coldkeypub_file=MagicMock(
+                exists_on_device=MagicMock(
+                    return_value=True # Wallet exists
+                ),
+                is_encrypted=MagicMock(
+                    return_value=False # Wallet is not encrypted
+                ),
+            ),
+            coldkeypub=MagicMock(
+                ss58_address=bittensor.Keypair.create_from_mnemonic(
+                        bittensor.Keypair.generate_mnemonic()
+                ).ss58_address
+            )
+        ),
+        MagicMock(
+            hotkey_file=MagicMock(
+                exists_on_device=MagicMock(
+                    return_value=True # Wallet exists
+                ),
+                is_encrypted=MagicMock(
+                    return_value=False # Wallet is not encrypted
+                ),
+            ),
+            hotkey=MagicMock(
+                ss58_address=bittensor.Keypair.create_from_mnemonic(
+                        bittensor.Keypair.generate_mnemonic()
+                ).ss58_address
+            )
+        )]):
+            config = self.config
+            config.wallet.path = 'tmp/walletpath'
+            config.wallet.name = 'mock_wallet'
+            config.subtensor.network = "mock"
+            config.no_prompt = True
+            config.subtensor._mock = True
+            config.command = "list"
+
+            cli = bittensor.cli(config)
+            with patch('os.walk', side_effect=[iter(
+                    [('/tmp/walletpath', ['mock_wallet'], [])] # 1 wallet dir
+            ),
+            iter(
+                [('/tmp/walletpath/mock_wallet/hotkeys', [], ['hk0'])] # 1 hotkey file
+            )]):
+                cli.run()
+
+    def test_list_no_wallet( self ):
+        # Mock IO for wallet
+        with patch('bittensor.Wallet.coldkeypub_file', MagicMock(
+            exists_on_device=MagicMock(
+                return_value=False # Wallet doesn't exist
+            )
+        )):
+            config = self.config
+            config.wallet.path = '/tmp/pytest'
```

Comment:
This might cause issues in parallel tests as this name is thrown around a lot. In a lazier implementation in a past PR, I just appended `time.time()` to the path. I've noticed this can still cause issues and collisions in the pathing.

Perhaps a better solution would be to give this a totally unique name (e.g. `/tmp/pytest_test_list_no_wallet`) if its completely encapsulated to this test.

Reply:
Could also add a function in `test_utils.py` that generates nonsense filepaths for us to use for testing (and keeps track of them akin to `get_random_unused_port`)

Reply:
> Perhaps a better solution would be to give this a totally unique name (e.g. `/tmp/pytest_test_list_no_wallet`) if its completely encapsulated to this test.

Went with this solution as the IO should be mocked anyway.